File: src/game/core/GameOver.java
package game.core;

/**
 * Ситуация окончания игры. Параметр ситуации - результат игры
 * (переменная типа GameResult)
 *
 * @author <a href="mailto:vladimir.romanov@gmail.com">Romanov V.Y.</a>
 */
public class GameOver extends Exception {
    /**
     *
     */
    private static final long serialVersionUID = 1L;

    public GameResult result;

    /**
     * Ситуация с результатом игры.
     *
     * @param result результатом игры.
     * @see GameResult
     */
    public GameOver(GameResult result) {
        this.result = result;
    }
}

File: GameCore/src/game/core/GameOver.kt
package game.core

/**
 * Ситуация окончания игры. Параметр ситуации - результат игры
 * (переменная типа GameResult)
 *
 * @author [Romanov V.Y.](mailto:vladimir.romanov@gmail.com)
 */
class GameOver(@JvmField val result: GameResult) : Exception() {
    companion object {
        private const val serialVersionUID = 1L
    }
}